Fix Issue#35: Add Redux functionality

kartikcode opened the pull request:
Resolves #35 .
Also added Redux functioning in sidebar component.(works fine)

kartikcode commented:
All the above edits have been done, kindly review it.

NilayMajorwar commented on src/redux/actions/index.js:
```diff
@@ -0,0 +1,26 @@
+  };
+};
+
+// Add action for unsubscribing the stream
+export const DelStream = (stream) => {
+  return {
+    type: "DEL_STREAM",
+    id: stream.id,
+    text: stream.name,
+  };
+};
+
+// Add action for changing the id of the active stream
+export const ActiveStream = (stream) => {
+  return {
+    type: "ACTIVE_STREAM",
+    id: stream.id,
+    text: stream.name,
+  };
+};
```
Please fix this too :sweat_smile:. Just take the stream ID as argument, and in the reducers, search through the streams list in store to get the corresponding stream object.

kartikcode commented:
Fixes done:
- Updated `activeStreamId` selector to `activeStream`
- Passing only stream ID in the actions.
- Updated selectors to iterate through the stream list and find the stream object to return the updated state
